version: '3.8'

services:
  avito-db:
    image: postgres:latest 
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_NAME}
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DB_USER} -d ${DB_NAME}"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - avito-network

  app:
    build: .
    depends_on:
      avito-db:
        condition: service_healthy
    environment:
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
      - POSTGRES_DB=${DB_NAME}
    ports:
      - "8080:8080"
    networks:
      - avito-network

  migrate:
    image: golang:alpine
    working_dir: /app
    volumes:
      - .:/app
    entrypoint: ["go", "run", "cmd/migrate/main.go", "up"]
    environment:
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
      - POSTGRES_DB=${DB_NAME}
    depends_on:
      avito-db:
        condition: service_healthy
    networks:
      - avito-network

volumes:
  postgres_data:

networks:
  avito-network:
    driver: bridge
